### DISCUSSION TITLE
Como validar los datos de conexión

### DISCUSSION THREAD

**Discussion by daniel89fg** [netflie/whatsapp-cloud-api#60] [General]
Hola.

Tengo el "number phone id" y el "access token" para realizar la conexión y envío de los mensajes. ¿Como podría verificar que son los datos correctos?. Estoy intentando crearme una función del tipo validate() y que devuelva un boleano, pero no me queda claro si sería la manera correcta.

Función de ejemplo:
```
public function validate(): bool
    {
        try {
            $response = $this->whatsapp_cloud_api->sendTextMessage('34555123456', 'API validation test');
            if ($response->isError()) {
                ToolBox::i18nLog()->error("Error validating API 1: ". $response->body());
                return false;
            }
        } catch (Exception $e) {
            ToolBox::i18nLog()->error("Error validating API 2: " . $e->getMessage());
            return false;
        }

        return true;
    }
```

Funciona, pero seguro podría existir otra forma mejor de hacerlo, ¿alguna idea?.

Gracias.

**Comment by aalbarca** [comment]
Hola @daniel89fg el código es correcto.

A mi particularmente me resulta raro una función llamada `validate` que necesite enviar un mensaje sobre una infraestructura (en este caso WhatsApp) para validar los datos introducidos en una API. ¿En qué contexto de tu aplicación necesitas validar si los datos de API configurados son correctos?

Saludos.

**Reply by daniel89fg** [reply]
En mi aplicación web el usuario registra los números de teléfono que necesite. Para ello debe guardar el id del teléfono (sacado de la app de Facebook) y el acceso token (sacado de la app de Facebook). Al guardar el teléfono en el sistema, la idea es poder conectar a la app y verificar que esos datos son correctos y no inventados. 

Actualmente lo he conseguido simplemente haciendo una llamada para obtener los datos de la empresa del teléfono. Cualquier llamada sería válida, solo es ver que conecta con esos datos.

**Comment by aalbarca** [comment]
Me he imaginado que irían por ahí los tiros. Efectivamente era lo que te iba a proponer, hacer la consulta que estás haciendo para obtener los datos asociados al teléfono introducido: https://developers.facebook.com/docs/whatsapp/cloud-api/reference/business-profiles

Lamentablemente este paquete en php no dispone de esta llamada, lo anoto para futuras versiones.

Saludos :)
